src/identity-utils.ts

```diff
@@ -0,0 +1,259 @@
+    if (!cache) {
+        return false;
+    }
+
+    const cacheKey: string = concatenateIdentities(
+        method,
+        proposedUserIdentities
+    );
+    const hashedKey = generateHash(cacheKey);
+
+    // if cache doesn't have the cacheKey, there is no valid cached identity
+    if (!cache.hasOwnProperty(hashedKey)) {
+        return false;
+    }
+    
+    // If there is a valid cache key, compare the expireTimestamp to the current time.
+    // If the current time is greater than the expireTimestamp, it is not a valid
+    // cached identity.
+    const expireTimestamp = cache[hashedKey].expireTimestamp;
+    
+    if (expireTimestamp < new Date().getTime()) {
+        return false;
+    } else {
+        return true;
+    }
+};
+
+export const getCachedIdentity = (
+    method: IdentityAPIMethod,
+    proposedUserIdentities: IKnownIdentities,
+    idCache: BaseVault<Dictionary<ICachedIdentityCall>>
+): ICachedIdentityCall | null => {
+    const cacheKey: string = concatenateIdentities(
+        method,
+        proposedUserIdentities
+    );
+    const hashedKey = generateHash(cacheKey);
+
+    const cache = idCache.retrieve();
+    const cachedIdentity = cache ? cache[hashedKey] : null;
+
+    return cachedIdentity;
+};
+
+// https://go.mparticle.com/work/SQDSDKS-6079
+export const createKnownIdentities = (
+    identityApiData: IdentityApiData,
+    deviceId: string
+): IKnownIdentities => {
+    const identitiesResult: IKnownIdentities = {};
+
+    if (
+        identityApiData?.userIdentities &&
+        isObject(identityApiData.userIdentities)
+    ) {
+        for (let identity in identityApiData.userIdentities) {
+            identitiesResult[identity] =
+                identityApiData.userIdentities[identity];
+        }
+    }
+    identitiesResult.device_application_stamp = deviceId;
+
+    return identitiesResult;
+};
+
+export const removeExpiredIdentityCacheDates = (idCache: BaseVault<Dictionary<ICachedIdentityCall>>) => {
+    const cache: Dictionary<ICachedIdentityCall> = idCache.retrieve() || {} as Dictionary<ICachedIdentityCall>;
+
+    idCache.purge();
+    
+    const currentTime: number = new Date().getTime();
+
+    // Iterate over the cache and remove any key/value pairs that are expired
+    for (let key in cache) {
+        if (cache[key].expireTimestamp < currentTime) {
+            delete cache[key];
+        }
+    };
+
+    idCache.store(cache);
+}
+
+export const tryCacheIdentity = (
+    mpInstance: MParticleWebSDK,
+    knownIdentities: IKnownIdentities,
+    idCache: BaseVault<Dictionary<ICachedIdentityCall>>,
+    parseIdentityResponse: IParseCachedIdentityResponse,
+    mpid: string,
+    callback: IdentityCallback,
+    identityApiData: IdentityApiData,
+    identityMethod: IdentityAPIMethod
+): boolean =>  {
+    if (
+        mpInstance._Helpers.getFeatureFlag(
+            Constants.FeatureFlags.CacheIdentity
+        )
+    ) {
```

Comment:
Based on our conversation earlier, you should do this check outside of this function. This way, there's no need to pass in the entire `mpInstance` object, and we reduce the number of arguments.